tests/testthat/_snaps/tree.md

```diff
@@ -0,0 +1,256 @@
+# Array-like indices can be shown or hidden
+
+    Code
+      nested_lists <- list(list(id = "a", val = 2), list(id = "b", val = 1, children = list(
+        list(id = "b1", val = 2.5), list(id = "b2", val = 8, children = list(list(id = "b21",
+          val = 4))))), list(id = "c", val = 8, children = list(list(id = "c1"), list(
+        id = "c2", val = 1))))
+      tree(nested_lists, index_unnamed = TRUE)
+    Output
+      [list]
+      ├─1:{list}
+      │ ├─id:"a"
+      │ └─val:2
+      ├─2:{list}
+      │ ├─id:"b"
+      │ ├─val:1
+      │ └─children:[list]
+      │   ├─1:{list}
+      │   │ ├─id:"b1"
+      │   │ └─val:2.5
+      │   └─2:{list}
+      │     ├─id:"b2"
+      │     ├─val:8
+      │     └─children:[list]
+      │       └─1:{list}
+      │         ├─id:"b21"
+      │         └─val:4
+      └─3:{list}
+        ├─id:"c"
+        ├─val:8
+        └─children:[list]
+          ├─1:{list}
+          │ └─id:"c1"
+          └─2:{list}
+            ├─id:"c2"
+            └─val:1
+    Code
+      tree(nested_lists, index_unnamed = FALSE)
+    Output
+      [list]
+      ├─{list}
+      │ ├─id:"a"
+      │ └─val:2
+      ├─{list}
+      │ ├─id:"b"
+      │ ├─val:1
+      │ └─children:[list]
+      │   ├─{list}
+      │   │ ├─id:"b1"
+      │   │ └─val:2.5
+      │   └─{list}
+      │     ├─id:"b2"
+      │     ├─val:8
+      │     └─children:[list]
+      │       └─{list}
+      │         ├─id:"b21"
+      │         └─val:4
+      └─{list}
+        ├─id:"c"
+        ├─val:8
+        └─children:[list]
+          ├─{list}
+          │ └─id:"c1"
+          └─{list}
+            ├─id:"c2"
+            └─val:1
+
+# Atomic arrays have sensible defaults with truncation added for longer than 10-elements
+
+    Code
+      tree(list(name = "vectored list", num_vec = 1:10, char_vec = letters))
+    Output
+      {list}
+      ├─name:"vectored list"
+      ├─num_vec:1,2,3,4,5,6,7,8,9,10
+      └─char_vec:"a","b","c","d","e","f","g","h","i","j",...(n = 26)
```

Comment:
What if this was `char_vec[26]`? i.e. always put the length on the left hand side.

Reply:
What happens if each individual string is very long?

Reply:
I ended up changing the vector printing to wrap in square brackets and then (if truncated) put the length at the end. Putting the length at the start felt like it interfered with ease-of-reading and meant that we needed to put it for every vector, which if they are short feels excessive. 

```
<list>
├─name: "vectored list"
├─num_vec: (n:10) 1, 2, 3, 4, 5, 6, 7, 8, 9, 10
└─char_vec: (n:26) "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", ...
```
vs
```
<list>
├─name: "vectored list"
├─num_vec: [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
└─char_vec: ["a", "b", "c", "d", "e", "f", "g", "h", "i", "j", ...] n:26
```

I'm not deadset in this viewpoint though.

Reply:
How about using something like `pillar::obj_sum()`:

```
├─name<chr [1]>: "vectored list"
├─num_vec<int [10]>: 1, 2, 3, 4, 5, 6, 7, 8, 9, 10
└─char_vec<chr [26]>: "a", "b", "c", "d", "e", "f", "g", "h", "i", "j", ...
```

I'd prefer not to use `[]` just because that'd be introducing a new convention which we hadn't used elsewhere.

Reply:
Oh I definitely like that better.

Reply:
Done with one exception. I feel like adding the same `<type [length]>` prefix for all the scalar atomics is a bit busier than necessary so I default to printing those as before; with an option to enable more verbose behavior.

Default keeps scalar atomics nice and clean (type's are still inferable with printing behavior)
```R
lobstr::tree(
  list(name = "vectored list", num_vec = 1:10)
)
#> <list>
#> ├─name: "vectored list"
#> └─num_vec<int [10]>: 1, 2, 3, 4, 5, 6, 7, 8, 9, 10
```
But the more verbose (and accurate to underyling structure) printing is available with an argument
```R
lobstr::tree(
  list(name = "vectored list", num_vec = 1:10), 
  hide_scalar_types = FALSE
)
#> <list>
#> ├─name<chr [1]>: "vectored list"
#> └─num_vec<int [10]>: 1, 2, 3, 4, 5, 6, 7, 8, 9, 10
```

 Do you think this is a fair compromise between terseness and conveying info about the element?